test/functional/p2p_tx_download.py

```diff
@@ -216,6 +216,24 @@ def test_preferred_inv(self):
         with p2p_lock:
             assert_equal(peer.tx_getdata_count, 1)
 
+    def test_txid_inv_delay(self):
+        self.log.info('Check that inv from a txid-relay peers are delayed by {} s'.format(TXID_RELAY_DELAY))
+        self.restart_node(0, extra_args=['-whitelist=noban@127.0.0.1'])
+        mock_time = int(time.time() + 1)
+        self.nodes[0].setmocktime(mock_time)
+        peer = self.nodes[0].add_p2p_connection(TestP2PConn(wtxidrelay=False))
+        # Add a second wtxid-relay connection otherwise TXID_RELAY_DELAY is waived in
+        # lack of wtxid-relay peers
+        self.nodes[0].add_p2p_connection(TestP2PConn(wtxidrelay=True))
+        peer.send_message(msg_inv([CInv(t=MSG_TX, h=0xff11ff11)]))
+        peer.sync_with_ping()
+        with p2p_lock:
+            assert_equal(peer.tx_getdata_count, 0)
+        self.nodes[0].setmocktime(mock_time + TXID_RELAY_DELAY)
+        peer.wait_until(lambda: peer.tx_getdata_count >= 1, timeout=1)
+        with p2p_lock:
+            assert_equal(peer.tx_getdata_count, 1)
```

Comment:
`wait until` right above wouldn't allow this line to happen if `tx_getdata_count == 0`. So we're checking it's NOT more than 1? Is it a worthy check?...

Reply:
Right, in fact it sounds other new functional tests added with #19988 have a redundant p2p lock tacking so added a new commit to remove this oversight. Unless @MarcoFalke they serve something else ?